yt_dlp/extractor/bild.py

```diff
@@ -19,19 +20,41 @@ class BildIE(InfoExtractor):
             'thumbnail': r're:^https?://.*\.jpg$',
             'duration': 196,
         }
-    }
+    }, {
+        'note': 'static MP4 and HLS',
+        'url': 'https://www.bild.de/video/clip/news-ausland/deftiger-abgang-vom-10m-turm-bademeister-sorgt-fuer-skandal-85158620.bild.htm',
+        'md5': 'fb0ed4f09c495d4ba7ce2eee0bb90de1',
+        'info_dict': {
+            'id': '85158620',
+            'ext': 'mp4',
+            'title': 'Der Sprungturm-Skandal',
+            'description': 'md5:709b543c24dc31bbbffee73bccda34ad',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'duration': 69,
+        }
+    }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
         video_data = self._download_json(
             url.split('.bild.html')[0] + ',view=json.bild.html', video_id)
 
+        formats = []
+        for src in video_data['clipList'][0]['srces']:
+            if src['type'] == 'application/x-mpegURL':
+                formats.extend(
+                    self._extract_m3u8_formats(
+                        src['src'], video_id, 'mp4',
+                        entry_protocol='m3u8_native', m3u8_id='hls', fatal=False))
+            elif src['type'] == 'video/mp4':
+                formats.append({'url': src['src'], 'format_id': 'http-mp4'})
```

Comment:
- Extract the formats non-fatally with `traverse_obj`: you can use branching (`...`) or filtering (`lambda _, v:`) so that it always returns a list. We can filter here for a truthy value to the `'src'` key
- `entry_protocol=m3u8_native` is the default, so we can omit
- Warn when an unsupported source type is skipped

```suggestion
        formats = []
        for src in traverse_obj(video_data, ('clipList', 0, 'srces', lambda _, v: v['src'])):
            src_type = src.get('type')
            if src_type == 'application/x-mpegURL':
                formats.extend(
                    self._extract_m3u8_formats(src['src'], video_id, 'mp4', m3u8_id='hls', fatal=False))
            elif src_type == 'video/mp4':
                formats.append({'url': src['src'], 'format_id': 'http-mp4'})
            else:
                self.report_warning(f'Skipping unsupported format type: "{src_type}"')
```

Reply:
> * Extract the formats non-fatally with `traverse_obj`: you can use branching (`...`) or filtering (`lambda _, v:`) so that it always returns a list. We can filter here for a truthy value to the `'src'` key

Why non-fatal? It's the only source of information this IE uses. Or do you want it to provoke a "no formats found" error instead, if the traversal fails?

> * `entry_protocol=m3u8_native` is the default, so we can omit

Okay. I was skimming through other extractors' uses of this function, and noticed this parameter quite often. I should have checked the source.

> * Warn when an unsupported source type is skipped

Absolutely. I noticed the missing `else` as well.

Reply:
> Why non-fatal? It's the only source of information this IE uses.

Filtering for `src['src']`/URLs (and thus filtering for dicts so we can also use `src.get('type')` non-fatally) is the important thing here; if we are extracting multiple formats we don't want to fatally error on one of them if another can be extracted successfully. 

> Or do you want it to provoke a "no formats found" error instead, if the traversal fails?

This is fine. A no formats error without any other warnings will tell us the same as a KeyError or TypeError would

> I was skimming through other extractors' uses of this function, and noticed this parameter quite often. 

It was not the default in youtube-dl, so what you see in other extractors is legacy code. Not a big deal at all, we can just tidy it up

Reply:
Thanks for the explanations. I have added a commit to the PR.

I also needed to revert the URL change (`html` to `html?`), as it broke the `split()` call. While the `.htm` URLs do work, they are a redirect anyway, and don't seem to appear in the wild.

(To be squashed, obviously.)